### PR TITLE
Change products by clicking on product

### PR DESCRIPTION
**Pull request by milofultz**
This PR allows the user to click on a card in the related items or outfit and go to/load that item.

### CODE REVIEW

File: client/src/components/ProductDetailPage.jsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ class ProductDetailPage extends React.Component {
       products: {
         // productId: { ... },
       },
-      yourOutfit: [],
+      yourOutfit: JSON.parse(localStorage.getItem('atelier-your-outfit')) || [],
```
Comment (NylonEric) [inline client/src/components/ProductDetailPage.jsx:31]: Is there a compelling reason to store the outfit in localStorage vs. in memory? Is this a substitute for sessioncookie/user state approach?

File: client/src/components/ProductDetailPage.jsx [hunk 2]
```diff
@@ -38,52 +38,69 @@ class ProductDetailPage extends React.Component {
...
+    this.getCurrentProductFromAPI(productId)
       .then(productInformation => {
-        updatedProducts[this.state.currentProductId] = productInformation;
+        updatedProducts[productId] = productInformation;
         this.setState({
-          products: updatedProducts
-        }, () => {
-          console.timeEnd('fetched current product');
+          products: {
+            ...this.state.products,
```
Comment (NylonEric) [inline client/src/components/ProductDetailPage.jsx:70]: Clever!

File: client/style.css [hunk 3]
```diff
@@ -752,6 +752,10 @@ h1{
   max-width: 240px;
   overflow: hidden;
 }
+.product__picture > .product__image,
+.product__picture > .product__no-picture-provided {
+  transition: all 0.2s ease-out;
```
Comment (NylonEric) [inline client/style.css:757]: NIce.

**Review by NylonEric — APPROVED** [review]
I like the stuff you're doing with the spread operator. I believe I understand the local storage choice was to make the outfit persist after refresh. That's probably a good thing to write a test for.

File: client/src/components/ProductDetailPage.jsx (hunk 1 above)
Comment (milofultz) [reply client/src/components/ProductDetailPage.jsx:31]: This just allows it to persist across sessions. Nothing we need to do, but it's neat to have it

**Pull request merged**